### PR TITLE
Gltf loader now shows which file is missing pre baked tangents

### LINKED ISSUE #11831: Improve gltf loader message about missing tangents
## What problem does this solve or what need does it fill?

When having many GLTF models in an app, if debug log level is turned on messages about missing vertex tangets might appear.
However it doesn't say _which_ model, so it's hard to take action.

## What solution would you like?

1. Inform which model is missing tangents
2. Also, if not too opiniated, mention that Blender is able to bake them (which I _think_ it can during gltf export). This would allow the user to know not only the problematic model, but a way to fix it too

## What alternative(s) have you considered?

Using a debugger to find out which model is problematic.

## Additional context

```
2024-02-12T07:33:22.774925Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:22.812107Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:22.833721Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:22.852133Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.302680Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.303184Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.304056Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.304817Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.309710Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.312120Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.318036Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.320884Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.326357Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.328809Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.336656Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.351425Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.366192Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.463666Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.468243Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.505040Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.527992Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.555847Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.560282Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.602515Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.683227Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.687447Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.723583Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.746370Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.772947Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.777070Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.816259Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.899092Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.904600Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.936691Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:23.970883Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.002366Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.025267Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.113704Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.118518Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.163051Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.197286Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.232813Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.267934Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.350247Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.353601Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.392365Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.425674Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.461521Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.496562Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.575401Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.589792Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.593798Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.630627Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.651874Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
2024-02-12T07:33:24.668495Z DEBUG bevy_gltf::loader: Missing vertex tangents, computing them using the mikktspace algorithm
```

### PR DESCRIPTION
**Pull request by andristarr**
# Objective

- Gltf loader now shows which file is missing pre baked tangents
- Fixes #11831

## Solution

- The file name is shown in the error message

- What changed as a result of this PR?
### Changed:
- Gltf loader now shows which file is missing pre baked tangents

- If this PR is a breaking change (relative to the last release of Bevy), describe how a user might need to migrate their code to support these changes
- Simply adding new functionality is not a breaking change.
- Fixing behavior that was definitely a bug, rather than a questionable design choice is not a breaking change.

### CODE REVIEW

**Comment by andristarr** [comment]
@torsteingrindvik Would you want to follow your suggested wording for the warning or are you okay if it is worded like this and doesn't show the whole duration of the file load, only the building of the missing tangents?

Additonally I do not feel strong if I have used the correct mappings to errors when getting the filename, but I wouldn't want to unwrap.

File: crates/bevy_gltf/src/loader.rs
Comment (torsteingrindvik) [inline crates/bevy_gltf/src/loader.rs:499]: Thinking about it, in my opinion it is better to remove the use of `Instant` and the `warn!` below, and rather wrap the block below which computes tangents in a `span!` which mentions the file name.

This way profilers can pick up how long it takes and aggregate it etc.

Also a warning would be a bit too heavy-handed anyway since some users might be ok with having them auto-computed.

File: crates/bevy_gltf/src/loader.rs
Comment (torsteingrindvik) [inline crates/bevy_gltf/src/loader.rs:496]:
```suggestion
                    "Missing vertex tangents for {}, computing them using the mikktspace algorithm. Consider using a tool such as Blender to pre-compute the tangents.", file_name
```

I'll leave it to the maintainers when it comes to deciding if mentioning a tool is too opinionated but for me this is ok.

File: crates/bevy_gltf/src/loader.rs
Comment (andristarr) [reply crates/bevy_gltf/src/loader.rs]: Did you have something like this in mind?

Comment (torsteingrindvik) [reply crates/bevy_gltf/src/loader.rs]: Yes- but perhaps using https://docs.rs/tracing/latest/tracing/span/struct.Span.html#method.in_scope
to avoid the explicit drop/exit

Comment (andristarr) [reply crates/bevy_gltf/src/loader.rs]: Gotcha!

File: crates/bevy_gltf/src/loader.rs
Comment (torsteingrindvik) [inline crates/bevy_gltf/src/loader.rs:181]: I tried this PR now locally, and I get logs such as:

```
2024-02-16T08:43:52.892063Z DEBUG bevy_gltf::loader: Missing vertex tangents for scene.gltf, computing them using the mikktspace algorithm. Consider using a tool such as Blender to pre-compute the tangents.
2024-02-16T08:43:52.920969Z DEBUG bevy_gltf::loader: Missing vertex tangents for scene.gltf, computing them using the mikktspace algorithm. Consider using a tool such as Blender to pre-compute the tangents.
2024-02-16T08:43:52.951979Z DEBUG bevy_gltf::loader: Missing vertex tangents for scene.gltf, computing them using the mikktspace algorithm. Consider using a tool such as Blender to pre-compute the tangents.
2024-02-16T08:43:52.981170Z DEBUG bevy_gltf::loader: Missing vertex tangents for scene.gltf, computing them using the mikktspace algorithm. Consider using a tool such as Blender to pre-compute the tangents.
2024-02-16T08:43:53.009484Z DEBUG bevy_gltf::loader: Missing vertex tangents for scene.gltf, computing them using the mikktspace algorithm. Consider using a tool such as Blender to pre-compute the tangents.
```

I think it's likely that other users also have a lot of non-unique GLTF names.
E.g. some asset stores often have such generic names when you buy models from them.

So perhaps using the load context path in full is better than just the file name.

File: crates/bevy_gltf/src/loader.rs
Comment (torsteingrindvik) [inline crates/bevy_gltf/src/loader.rs:499]: As far as I can tell every other span in Bevy is `info_span!` so I suggest using that.

**Review by torsteingrindvik — APPROVED** [review]

**Comment by torsteingrindvik** [comment]
image

Thank you!

The log message now states exactly which model is being generated tangents for, and tracy shows that my suspicion was correct: There is way too much time being spent in generating these.

File: crates/bevy_gltf/src/loader.rs
Comment (andristarr) [reply crates/bevy_gltf/src/loader.rs]: @alice-i-cecile if you have a few free cycles please let us know about your opinion!

Comment (alice-i-cecile) [reply crates/bevy_gltf/src/loader.rs]: Happy to endorse blender here.

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**
